x-pack/plugins/asset_manager/server/plugin.ts

```diff
@@ -15,26 +15,48 @@ import {
   PluginConfigDescriptor,
   Logger,
 } from '@kbn/core/server';
+
 import { upsertTemplate } from './lib/manage_index_templates';
+import { runImplicitCollection } from './lib/implicit_collection';
 import { setupRoutes } from './routes';
 import { assetsIndexTemplateConfig } from './templates/assets_template';
 
 export type AssetManagerServerPluginSetup = ReturnType<AssetManagerServerPlugin['setup']>;
-export interface AssetManagerConfig {
-  alphaEnabled?: boolean;
-}
+
+const configSchema = schema.object({
+  alphaEnabled: schema.maybe(schema.boolean()),
+  implicitCollection: schema.maybe(
+    schema.object({
+      enabled: schema.maybe(schema.boolean({ defaultValue: true })),
+      interval: schema.duration({ defaultValue: '5m' }),
+      input: schema.object({
+        hosts: schema.string(),
+        username: schema.string(),
+        password: schema.string(),
+      }),
```

Comment:
Using the default client with `kibana_system` user errors out when writing documents and we can't pass `elastic` as user anymore so I ended adding the input as a workaround

```
action [indices:data/write/bulk[s]] is unauthorized for user [kibana_system] with effective roles [kibana_system] on indices [assets-container-default], this action is granted by the index privileges [create_doc,create,delete,index,write,all]
```